docs/src/model_simulation/ode_simulation_performance.md

```diff
@@ -172,6 +172,26 @@ Finally, we note that when using preconditioners with a matrix-free method (like
 Generally, the use of preconditioners is only recommended for advanced users who are familiar with the concepts. However, for large systems, if performance is essential, they can be worth looking into.
 
 
+## [Elimination of system conservation laws](@id ode_simulation_performance_conservation_laws)
+Previously, we have described how Catalyst, when it generates ODEs, is able to [detect and eliminate conserved quantities](@ref network_analysis_conservation_laws). In certain cases, doing this can improve performance. E.g. in the following example we will eliminate the single conserved quantity in a [two-state model](@ref basic_CRN_library_two_states). This results in a differential algebraic equation with a single differential equation and a single algebraic equation (as opposed to two differential equations). Conservation laws can be eliminated by providing the `remove_conserved = true` option to `ODEProblem`:
+```@example ode_simulation_performance_conservation_laws
+using Catalyst, OrdinaryDiffEq
+
+# Declare model.
+rs = @reaction_network begin
+    (k₁,k₂), X₁ <--> X₂
+end
+
+# Generate and simulate and ODE (while eliminating any conserved quantities).
+u0 = [:X₁ => 2.0, :X₂ => 3.0]
+ps = [:k₁ => 1.0, :k₂ => 2.0]
+oprob = ODEProblem(rs, u0, (0.0, 10.0), ps; remove_conserved = true)
+sol = solve(oprob)
+nothing # hide
+```
+Conservation law elimination should never impact performance negatively. However, its effect on performance can be limited, and is very model-dependent. It is primarily important when performing simulation with [implicit solvers](@ref ode_simulation_performance_stiffness) (as these requires computing system Jacobians, and conservation law eliminations typically transform these from singular Jacobians to non-singular Jacobians).
```

Comment:
I did some light tests, saw a (very  minor but at least noticeable) speed up when I had a model with several conservation laws and implicit solver.

Reply:
At some point it would be nice to do more exhaustive tests.

Reply:
Conservation law elimination can prove **very** beneficial for models with steady-state simulations where we want to compute gradients. Briefly, if the ODE Jacobian is invertiable, we can obtain the gradient via a simple linear-solve, they perform some benchmarks on this here https://arxiv.org/pdf/2405.16524.

Reply:
Interesting! Currently, Catalyst does not default to `remove_conserved = true` when it creates ODEs (there are some discussions to do so, but right now there are some niche annoyances which make me reluctant).

However, it might make sense for PEtab to default to using this feature? Since PEtab handles stuff internally, e.g. that eliminated variables are not plotted by default in ODE simulations should not be a problem.

Reply:
Ahh, I agree. When I have updated PEtab to MTKv9 I have some models where it would be interesting to try this out.

Reply:
excellent, happy to help (and for the conservation law feature to get more used. I have no part in implementing it, but it is really cool)